nym-vpn-apple/Services/Sources/Services/NymLogger/FileLogHandler.swift

```diff
@@ -63,10 +63,20 @@ public class FileLogHandler: LogHandler {
         let data = Data(logLine.utf8)
 
         ioQueue.async {
-            if self.fileHandle == nil, let logFileURL = FileLogHandler.logFileURL {
-                self.fileHandle = try? FileHandle(forWritingTo: logFileURL)
-            }
-            try? self.fileHandle?.write(contentsOf: data)
+            do {
+                if self.fileHandle == nil, let logFileURL = FileLogHandler.logFileURL {
+                    self.fileHandle = try FileHandle(forWritingTo: logFileURL)
+                }
+                try self.fileHandle?.write(contentsOf: data)
+            } catch CocoaError.fileNoSuchFile {
+                guard let logFileURL = FileLogHandler.logFileURL,
+                      !FileManager.default.fileExists(atPath: logFileURL.relativePath)
+                else {
+                    return
+                }
+                FileManager.default.createFile(atPath: logFileURL.relativePath, contents: nil, attributes: nil)
+                self.fileHandle = nil
```

Comment:
You need to open log file after creating it and log the line. Otherwise you basically drop the logged line.

Reply:
Maybe we should create the file beforehand to avoid duplicating the code?


```swift
if self.fileHandle == nil, let logFileURL = FileLogHandler.logFileURL {
    if !<file exists?> {
        FileManager.default.createFile(atPath: logFileURL.path(false), contents: nil, attributes: nil)
    }
    self.fileHandle = try? FileHandle(forWritingTo: logFileURL)
    try self.fileHandle?.write(contentsOf: data)
}
```

Reply:
Maybe we should create the file beforehand to avoid duplicating the code?


```swift
if self.fileHandle == nil, let logFileURL = FileLogHandler.logFileURL {
    if !<file exists?> {
        FileManager.default.createFile(atPath: logFileURL.path(false), contents: nil, attributes: nil)
    }
    self.fileHandle = try? FileHandle(forWritingTo: logFileURL)
    try self.fileHandle?.write(contentsOf: data)
}
```